x-pack/plugins/cloud_security_posture/public/pages/rules/rules_counters.tsx

```diff
@@ -0,0 +1,319 @@
+import { useCspIntegrationLink } from '../../common/navigation/use_csp_integration_link';
+import { useNavigateFindings } from '../../common/hooks/use_navigate_findings';
+import { cloudPosturePages } from '../../common/navigation/constants';
+import {
+  CSPM_POLICY_TEMPLATE,
+  KSPM_POLICY_TEMPLATE,
+  RULE_FAILED,
+  RULE_PASSED,
+} from '../../../common/constants';
+import { statusColors } from '../../common/constants';
+import { useCspBenchmarkIntegrationsV2 } from '../benchmarks/use_csp_benchmark_integrations';
+import { DASHBOARD_COUNTER_CARDS } from '../compliance_dashboard/test_subjects';
+import { CspCounterCard } from '../../components/csp_counter_card';
+import { useKibana } from '../../common/hooks/use_kibana';
+
+const EvaluationPieChart = ({ failed, passed }: { failed: number; passed: number }) => {
+  const {
+    services: { charts },
+  } = useKibana();
+
+  return (
+    <Chart size={{ height: 30, width: 30 }}>
+      <Settings
+        theme={[
+          {
+            partition: {
+              linkLabel: { maximumSection: Infinity, maxCount: 0 },
+              outerSizeRatio: 0.75,
+              emptySizeRatio: 0.7,
+            },
+          },
+        ]}
+        baseTheme={charts.theme.useChartsBaseTheme()}
+      />
+      <Partition
+        id={'evaluation-pie-chart'}
+        data={[
+          {
+            label: i18n.translate('xpack.csp.rulesPage.evaluationPieChart.failedTitle', {
+              defaultMessage: 'Failed',
+            }),
+            value: failed,
+          },
+          {
+            label: i18n.translate('xpack.csp.rulesPage.evaluationPieChart.passedTitle', {
+              defaultMessage: 'Passed',
+            }),
+            value: passed,
+          },
+        ]}
+        valueGetter="percent"
+        valueAccessor={(d) => d.value}
+        layout={PartitionLayout.sunburst}
+        layers={[
+          {
+            groupByRollup: (d: { label: string }) => d.label,
+            shape: {
+              fillColor: (label) =>
+                label.toLowerCase() === RULE_PASSED.toLowerCase()
+                  ? statusColors.passed
+                  : statusColors.failed,
+            },
+          },
+        ]}
+      />
+    </Chart>
+  );
+};
+
+export const RulesCounters = () => {
+  const { http } = useKibana().services;
+  const rulesPageParams = useParams<{ benchmarkId: string; benchmarkVersion: string }>();
+  const getBenchmarks = useCspBenchmarkIntegrationsV2();
+  const navToFindings = useNavigateFindings();
+  const cspmIntegrationLink = useCspIntegrationLink(CSPM_POLICY_TEMPLATE) || '';
+  const kspmIntegrationLink = useCspIntegrationLink(KSPM_POLICY_TEMPLATE) || '';
+
+  const benchmarkRulesStats = getBenchmarks.data?.items.find(
+    (benchmark) =>
+      benchmark.id === rulesPageParams.benchmarkId &&
+      benchmark.version === rulesPageParams.benchmarkVersion
+  );
+
+  if (!benchmarkRulesStats) {
+    return <></>;
+  }
```

Comment:
what about instead of returning the Fragment, handle separate the loading state using the `defaultLoadingRenderer` render function and the no data state using the `defaultNoDataRenderer` utilities in the cloud posture page component?

```js
 if (getBenchmarks.isLoading) {
    return defaultLoadingRenderer();
  }
if (!benchmarkRulesStats) {
    return defaultNoDataRenderer();
  }
```

alternatively, we could also wrap the entire page with the `<CloudPosturePage/>` component, so there are also other checks out of the box as well

Reply:
yeah the rule page itself is already using `CloudPosturePage` with the relevant query so we display a loader until this call is done.

Truth is this section only exist so the types for this file will function without assertions, but we never actually hit this branch.

and regarding the no data render, this is a different since this is just a component a part of the page. the no data render is meant for full page. here we have a component above the table that has an empty state so its handled individually